### ISSUE TITLE
Presets: straddle, strangle, iron condor, credit spread, iron fly

### ISSUE THREAD

**Issue by lalitkarthik** [lalitkarthik/convex-hedge-payoff#30]
## Parent

#23

## What to build

One click for a structure that would otherwise take four.

A trader picks a straddle, strangle, iron condor, credit spread or iron fly from a short list, and the Legs appear selected in the Chain ready to analyse — rather than being hunted down one row at a time.

**This adds no engine code, and that is the point.** A Preset is a function returning a list of Legs. Every metric is already computed from a list of Legs with no knowledge of what it is called, so an iron condor goes through exactly the same path a single Leg does. If this ticket turns out to need a change in the Strategy layer, something earlier was built wrong.

A Preset produces a Strategy; it is not a kind of Strategy. The trader can edit the Legs afterwards like any others, and the result stops matching the Preset without anything breaking.

## Acceptance criteria

- [ ] Straddle, strangle, iron condor, credit spread and iron fly are each selectable and produce the correct Legs
- [ ] Each Preset's Legs appear selected in the Chain and can be edited or removed afterwards
- [ ] Analysing a Preset produces the same result as selecting its Legs by hand
- [ ] The iron condor's four Legs produce Max Profit, Max Loss and both Breakevens with no branching on leg count
- [ ] No change is required in the Strategy aggregation layer to support any Preset
- [ ] Behaviour is asserted at the HTTP boundary

## Blocked by

- Ticket 5 — Chain table and multi-Leg Strategies

**Comment by MeghavPradeep** [comment]
**The backend half is open as #40**, CI green. Leaving this issue open — the picker UI, marking the Legs selected in the Chain, and editing/removing them there all need #25 first.

## The tripwire held

This ticket's real criterion is architectural, so here is the evidence rather than an assurance:

```
 src/payoff/api.py          |  40 +++-
 src/payoff/chain.py        |  14 ++
 src/payoff/models.py       |  13 ++
 src/payoff/presets.py      | 130 +++++++++++
 tests/test_api_presets.py  | 238 ++++++++++++++++++
```

**`strategy.py` does not appear.** Five structures of one, two and four Legs go down the same path a naked call does.

A Preset returns its Legs as **requests**, so choosing a Preset and picking the same rows off the Chain are the same operation rather than two paths that agree — the test asserts the two responses are *equal*, not close.

## Verified at the anchor moment

Centre 25100 (the quoted strike nearest the 25100.25 spot), width 200, per unit:

| Preset | Legs | max profit | max loss | breakevens | r/r |
|---|---|---|---|---|---|
| straddle | +25100CE +25100PE | `null` | −682.35 | 24417.65 / 25782.35 | `null` |
| strangle | +25300CE +24900PE | `null` | −503.05 | 24396.95 / 25803.05 | `null` |
| iron condor | −24900PE +24700PE −25300CE +25500CE | 141.35 | −58.65 | 24758.65 / 25441.35 | 2.410 |
| credit spread | −25100PE +24900PE | 71.95 | −128.05 | 25028.05 | 0.562 |
| iron fly | −25100CE −25100PE +25300CE +24900PE | 179.30 | −20.70 | 24920.70 / 25279.30 | 8.662 |

## On "a Preset produces a Strategy; it is not a kind of Strategy"

The sharpest version is asserted: **remove the bought call wing from the condor and the bounded loss becomes an Unbounded one** — `max_loss: null`, rather than the engine refusing a structure that no longer has a name. That test passed the moment it was written, with no implementation change. That is the architectural claim demonstrating itself rather than being asserted.

**Strategy Label is deliberately not built.** A Preset goes Legs-outward; a Label goes Legs-inward, it is in #23's out-of-scope list, and building the second while building the first is how a shape name ends up load-bearing.

## Two decisions worth a second opinion

1. **`direction` is ignored by the iron condor, credit spread and iron fly.** Those three are defined by their own internal signs — reversing them gives a long condor or a debit spread, which are *different trades*, not the same one held the other way. The straddle and strangle take `direction` because there they genuinely are. Happy to parameterise all five if you disagree.
2. **An unknown preset name returns 404** with a plain `{"detail": ...}` body. The status is here because a 500 would claim the server broke and it did not — but **#31 owns the body** and should restate it with a machine-readable code.

**Comment by MeghavPradeep** [comment]
Cross-reference from #44 — no change to `presets.py`, and this issue stays open.

#44 adds the four vertical spreads to the payoff notebook: bull call and bear put (debit), bear call and bull put (credit). The **bull put spread** is the same shape `presets.credit_spread` builds — sell the put at the centre, buy one `width` below — arrived at independently from the chain rather than from the Preset.

Two things from it that bear on this module:

**The credit/debit symmetry is arithmetic, not two cases.** A vertical's caps are one formula with the sign of `net` flipped:

```
max profit - max loss = wL
credit:  max profit = credit,  max loss  = credit - wL
debit:   max loss   = -debit,  max profit = wL - debit
```

The docstring on `credit_spread` says reversing the signs "is a debit spread, a different trade". True as a trade; worth noting it is the *same* computation, so nothing downstream needs to branch on it. The notebook asserts both directions against the engine at two different anchors.

**`DEFAULT_WIDTH` assumes the strike exists.** `centre ± width` is only tradable if that strike is quoted. At the 12:00 snapshot **25,400 CE is not**, so a spread or wing nominally 300 points out actually lands 400 out, and any cap computed from the nominal width is wrong. The notebook now places outer legs at the *nearest quoted strike at least `w` away* and reads the width back off the strikes it actually used.

That is a `LegRequest`-construction concern rather than a Preset-shape one, and the API may well be right to hand back the nominal strike and let the chain reject it. Flagging it here so it is not rediscovered — it is the kind of thing that produces a plausible-looking max loss that is off by a third.

The tripwire still holds: #44 touches no source file, and `strategy.py` remains unchanged.

**Comment by MeghavPradeep** [comment]
## Audit — the server half is merged

`src/payoff/presets.py` and `tests/test_api_presets.py` are on `main`, delivered by **#40**. `GET /presets` and `GET /presets/{name}` both serve. The **UI half remains** and is what this ticket now covers.

### One bug to settle before the UI

`credit_spread`, `iron_condor` and `iron_fly` all accept a `direction` argument and **silently ignore it** (`presets.py:62`, `:76`, `:92`). Only `straddle` and `strangle` honour it.

Either honour it in all five or remove the control for those three — but do not ship a direction toggle that does nothing on three of five presets.

### One trap

`iron_condor` reaches `centre ± 2*width` = ±400 by default. At a thin moment that falls outside the quoted strikes and raises `StrikeNotQuoted`, which has no handler and surfaces as a **500**. So a preset button can 500 the app until #31 lands.

### One consequence of #51

Presets centre on `chain.at_the_money()`, which changes from nearest-**spot** to nearest-**forward**. The ATM strike moves 25,100 → 25,200, so every preset's legs move with it. Existing preset tests will need their expected strikes updated.

Spec: `convex-hedge-plans/web-platform-spec/02-TICKETS.md`.
